api/src/main/java/org/datacleaner/api/WSStatelessComponent.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.api;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Components with the {@link org.datacleaner.api.WSStatelessComponent} annotation are provided
+ * in a web service for components.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.TYPE })
+public @interface WSStatelessComponent {
```

Comment:
I still feel this is not correct though. I can agree to the fact that some transformers should maybe not be exposed this way. But that's a more general characteristic of some (quite few) transformers and such characteristic should be explicitly annotated, not assumed. In other words: I think the default here is wrong. By default we should expose all components, but we could add an annotation for excluding particular components (separate story, please don't add any more to this PR since it's too big - and I have quite some comments on such an annotation also, since we should use the same statefulness flag in other areas than WS specific).